libraries/ms-common/src/x86/adc.c

```diff
@@ -176,32 +178,200 @@ StatusCode adc_read_converted(AdcChannel adc_channel, uint16_t *reading) {
   return STATUS_CODE_OK;
 }
 
-// the following functions are wrappers over the legacy AdcChannel API dealing with GpioAddresses
-// instead
-StatusCode adc_set_channel_pin(GpioAddress address, bool new_state) {
-  AdcChannel channel;
-  status_ok_or_return(adc_get_channel(address, &channel));
-  return adc_set_channel(channel, new_state);
+#else  // #ifndef MPXE
+#include <stddef.h>
+#include <stdlib.h>
+
+#include "adc.pb-c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "store.h"
+#include "stores.pb-c.h"
+
+#define ADC_RETURNED_VOLTAGE_RAW 2500
+#define ADC_CONTINUOUS_CB_FREQ_MS 50
+#define ADC_TEMP_RETURN 293
+#define ADC_VDDA_RETURN 3300
+
+typedef struct AdcInterrupt {
+  AdcCallback callback;
+  void *context;
+  uint16_t reading;
+} AdcInterrupt;
+
+static MxAdcStore s_store = MX_ADC_STORE__INIT;
+
+static AdcInterrupt s_adc_interrupts[NUM_ADC_CHANNELS];
+
+static bool s_active_channels[NUM_ADC_CHANNELS];
+
+static void prv_export() {
```

Comment:
since this library is purely for sensing hardware values and not for setting them, we never need to export data to python